CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc

```diff
@@ -89,18 +98,29 @@ void CaloTPGTranscoderULUT::loadHCALCompress(HcalLutMetadata const& lutMetadata,
 	double granularity =  meta->getLutGranularity(); 
 
 	if(isHBHE){
-	    double factor = nominal_gain_ / cosh_ieta * granularity;
-	    LUT tpg = outputLUT_[index][0];
-	    int low = 0;
-	    for (unsigned int i = 0; i < getOutputLUTSize(id); ++i){
-		if (outputLUT_[index][i] != tpg){
-		   unsigned int mid = (low + i)/2; 
-		   hcaluncomp_[index][tpg] = (tpg == 0 ? low : factor * mid);
-		   low = i;
-		   tpg = outputLUT_[index][i];
-		}
-	    }
-	    hcaluncomp_[index][tpg] = factor * low;
+           if (allLinear_) {
+              LUT tpg = outputLUT_[index][0];
+              hcaluncomp_[index][tpg] = 0;
+              for (unsigned int i = 0; i < getOutputLUTSize(id); ++i){
```

Comment:
if you store `getOutputLUTSize(id)` in a variable, you do not have to compute it in each loop iteration

Reply:
Sorry, I overlooked that that is already a variable…